src/main/java/com/google/opensesame/servlets/MentorsServlet.java

```diff
@@ -60,4 +74,55 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(jsonMentors);
   }
+
+  // This function prints an error message to the mentor form page if invalid input is found.
+  public void formError(HttpServletRequest request, HttpServletResponse response, String error)
+      throws ServletException, IOException {
+    String message =
+        "<ul><li class='ml-3 display-4 font-weight-bold text-danger'>" + error + "</li></ul>";
+    response.setContentType("text/html");
+    PrintWriter pw = response.getWriter();
+    pw.print(message);
+    RequestDispatcher rd = request.getRequestDispatcher("mentor_form.html");
+    rd.include(request, response);
+    pw.close();
+    return;
+  }
```

Comment:
Can we have a call about this? I think it's important that we stick to our convention of JSON error responses, and I think there's any easy fix here. Take a look at the [fetch error handler](https://github.com/googleinterns/step22-2020/blob/master/src/main/webapp/js/fetch_handler.js) for more information on the convention I'm talking about. Instead of returning HTML, you can use JSON to return the error and add it to the HTML on the client side.

Reply:
Didn't realize the full context of this as part of a fetch call being made directly from a form. That's my bad. This is totally fine!